materialize-postgres-rc/driver.go

```diff
@@ -117,6 +135,31 @@ func newPostgresDriver() pm.DriverServer {
 			}
 			var metaSpecs, metaCheckpoints = sql.MetaTables(metaBase)
 
+			// If SSH Endpoint is configured, then try to start a tunnel before establishing connections
+			if cfg.NetworkTunnel != nil && cfg.NetworkTunnel.SshForwarding != nil && cfg.NetworkTunnel.SshForwarding.SshEndpoint != "" {
+				host, port, err := net.SplitHostPort(cfg.Address)
+				if err != nil {
+					return nil, fmt.Errorf("splitting address to host and port: %w", err)
+				}
+
+				var sshConfig = &networkTunnel.SshConfig{
+					SshEndpoint: cfg.NetworkTunnel.SshForwarding.SshEndpoint,
+					PrivateKey:  []byte(cfg.NetworkTunnel.SshForwarding.PrivateKey),
+					ForwardHost: host,
+					ForwardPort: port,
+					LocalPort:   "5432",
+				}
+				var tunnel = sshConfig.CreateTunnel()
+
+				// FIXME/question: do we need to shut down the tunnel manually if it is a child process?
+				// at the moment tunnel.Stop is not being called anywhere, but if the connector shuts down, the child process also shuts down.
+				err = tunnel.Start()
```

Comment:
On one hand, it seems like it would be good to call `tunnel.Stop` to clean up the temp key file. But since its running in a container with temporary storage it doesn't seem like it would be strictly necessary so I can't see why it would be required to shut it down manually.